### DISCUSSION TITLE
MobileFormatter - DateTime.Kind lost after deserialization

### DISCUSSION THREAD

**Discussion by GlenYappa** [MarimerLLC/csla#3362] [Questions]
I am in the process of upgrading a large business object library from Csla 4 to Csla 6.2.2 while at the same time moving from .Net Framework 4.7.2 to .Net Standard 2.0. One issue that I am having is DateTime properties are losing the DateTime.Kind when deserialized. In the example below I have a simple ReadOnly csla object with a nullable Datetime and I am serializing/deserializing with the MobileFormatter directly just for test purposes. You can see that after deserialization the Kind value is Unspecified. Is this a known issue? 

	var t = TestHarness.App.DataPortalFactory.GetPortal<TestHarness.DateTimeTest>().Create();
	t.ID = Guid.NewGuid();
	t.TestDate = DateTime.Now;

	Console.WriteLine(t);

	var formatter = new Csla.Serialization.Mobile.MobileFormatter(TestHarness.App.AppContext);

	byte[] bytes = new byte[0];
	using (MemoryStream stream = new MemoryStream())
	{
		formatter.Serialize(stream, t);
		bytes = stream.ToArray();
	}

	using (MemoryStream stream = new MemoryStream(bytes))
	{
		t = (TestHarness.DateTimeTest)formatter.Deserialize(stream);
	}

	Console.WriteLine(t);

ID = ba5ddc50-75ea-4b7d-b656-7fc12faab73e
TestDate = 6/2/2023 8:59:27 AM
DateTime Kind = Local

ID = ba5ddc50-75ea-4b7d-b656-7fc12faab73e
TestDate = 6/2/2023 8:59:27 AM
DateTime Kind = Unspecified

Thanks

**Comment by rockfordlhotka** [comment]
This is not a known issue. I suspect that the writer(s) and readers in `Csla.Serialization.Mobile` need to be enhanced to serialize more than just the ticks value for a `DateTime`.

https://github.com/MarimerLLC/csla/blob/91921c07ac5679f986147554b9f8ddf350bbe2c0/Source/Csla/Serialization/Mobile/CslaBinaryWriter.cs#L227

**Reply by rockfordlhotka** [reply]
I've filed this as a backlog issue - #3363

**Reply by GlenYappa** [reply]
thanks
